core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadWriteLDAPUserStoreManager.java

```diff
@@ -1352,82 +1352,93 @@ public void doUpdateRoleListOfUser(String userName, String[] deletedRoles, Strin
+                                            mainDirContext,
+                                            searchBase);
+                            SearchResult resultedGroup = null;
+                            String groupDN = null;
+                            if (groupResults.hasMore()) {
+                                resultedGroup = groupResults.next();
+                                groupDN = resultedGroup.getName();
+                            }
+                            if (resultedGroup != null && isUserInRole(userNameDN, resultedGroup)) {
+                                this.modifyUserInRole(userNameDN, groupDN, DirContext.REMOVE_ATTRIBUTE,
                                         searchBase);
-                        SearchResult resultedGroup = null;
-                        String groupDN = null;
-                        if (groupResults.hasMore()) {
-                            resultedGroup = groupResults.next();
-                            groupDN = resultedGroup.getName();
-                        }
-                        this.modifyUserInRole(userNameDN, groupDN, DirContext.REMOVE_ATTRIBUTE,
-                                searchBase);
+                            } else {
+                                errorMessage =
+                                        "User: " + userName + " does not belongs to role: " +
+                                                deletedRole;
+                                throw new UserStoreException(errorMessage);
+                            }
 
-                        JNDIUtil.closeNamingEnumeration(groupResults);
+                            JNDIUtil.closeNamingEnumeration(groupResults);
 
-                        // need to update authz cache of user since roles
-                        // are deleted
-                        userRealm.getAuthorizationManager().clearUserAuthorization(userName);
+                            // need to update authz cache of user since roles
+                            // are deleted
+                            userRealm.getAuthorizationManager().clearUserAuthorization(userName);
 
-                    } else {
-                        errorMessage = "The role: " + deletedRole + " does not exist.";
-                        throw new UserStoreException(errorMessage);
+                        } else {
+                            errorMessage = "The role: " + deletedRole + " does not exist.";
+                            throw new UserStoreException(errorMessage);
+                        }
                     }
                 }
             }
             if (newRoles != null && newRoles.length != 0) {
 
                 for (String newRole : newRoles) {
 
-                    LDAPRoleContext context = (LDAPRoleContext) createRoleContext(newRole);
-                    newRole = context.getRoleName();
-                    String searchFilter = context.getSearchFilter();
+                    if (StringUtils.isNotEmpty(newRole)) {
```

Comment:
Is this check necessary here as well. Unless the array includes empty strings this will not happen